File: stock-check-service/src/main/java/com/chrisbaileydeveloper/stockcheckservice/service/StockCheckService.java
package com.chrisbaileydeveloper.stockcheckservice.service;

import com.chrisbaileydeveloper.stockcheckservice.dto.StockCheckResponse;
import com.chrisbaileydeveloper.stockcheckservice.repository.StockCheckRepository;
import lombok.RequiredArgsConstructor;
import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
@Slf4j
public class StockCheckService {

    private final StockCheckRepository stockCheckRepository;

    @Transactional(readOnly = true)
    @SneakyThrows
    public List<StockCheckResponse> isInStock(List<String> skuCode) {
        log.info("Checking Stock for {}", skuCode);
        return stockCheckRepository.findBySkuCodeIn(skuCode).stream()
                .map(stockCheck ->
                        StockCheckResponse.builder()
                                .skuCode(stockCheck.getSkuCode())
                                .isInStock(stockCheck.getQuantity() > 0)
                                .build()
                ).toList();
    }
}

File: stock-check-service/src/main/java/com/chrisbaileydeveloper/stockcheckservice/service/StockCheckService.kt
package com.chrisbaileydeveloper.stockcheckservice.service

import com.chrisbaileydeveloper.stockcheckservice.dto.StockCheckResponse
import com.chrisbaileydeveloper.stockcheckservice.repository.StockCheckRepository
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
open class StockCheckService(private val stockCheckRepository: StockCheckRepository) {

    private val log = LoggerFactory.getLogger(StockCheckService::class.java)

    @Transactional(readOnly = true)
    open fun isInStock(skuCode: List<String>): List<StockCheckResponse> {
        log.info("Checking Stock for {}", skuCode)
        return stockCheckRepository.findBySkuCodeIn(skuCode).map { stockCheck ->
            StockCheckResponse(
                skuCode = stockCheck.skuCode,
                isInStock = stockCheck.quantity > 0
            )
        }
    }
}
